pysnark/boolean.py

```diff
@@ -91,12 +113,18 @@ def __rtruediv__(self, other):
     def __neg__(self):
         """
         Performs an arithmetic negation
+        Returns a LinComb
         Costs 0 constraints
         """
         return -self.lc
     
     def __invert__(self):
-        return ConstVal(1) - self.lc
+        """
+        Performs a logical not
+        Returns a LinCombBool
+        Costs 1 constraint

```

Comment:
This costs zero constraints, right?

Reply:
This costs 1 constraint because we create a `LinCombBool`. The constraint ensures the witness variable is either a 1 or a 0.